### PR TITLE
Allow to set app domains

### PR DESCRIPTION
**Pull request by raphink**
This allows to set domains for each application.

Depends on #377

### CODE REVIEW

File: argocd/app-of-apps/values.tmpl.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ argo-cd:
       enabled: true
   server:
     config:
-      url: https://argocd.apps.${cluster_name}.${base_domain}
+      url: ${argocd.domain}
```
Comment (mcanevet) [inline argocd/app-of-apps/values.tmpl.yaml:65]: I guess the scheme is mandatory.

File: argocd/app-of-apps/values.tmpl.yaml [hunk 2]
```diff
@@ -130,13 +130,13 @@ keycloak:
       secret: ${oidc.client_secret}
       redirectUris:
         - https://argocd.apps.${base_domain}/auth/callback
-        - https://argocd.apps.${cluster_name}.${base_domain}/auth/callback
+        - ${argocd.domain}/auth/callback
```
Comment (mcanevet) [inline argocd/app-of-apps/values.tmpl.yaml:133]: same here

File: argocd/app-of-apps/values.tmpl.yaml [hunk 3]
```diff
@@ -130,13 +130,13 @@ keycloak:
...
         - https://grafana.apps.${base_domain}/login/generic_oauth
-        - https://grafana.apps.${cluster_name}.${base_domain}/login/generic_oauth
+        - ${grafana.domain}/login/generic_oauth
```
Comment (mcanevet) [inline argocd/app-of-apps/values.tmpl.yaml:135]: same here

File: argocd/app-of-apps/values.tmpl.yaml [hunk 4]
```diff
@@ -130,13 +130,13 @@ keycloak:
...
         - https://prometheus.apps.${base_domain}/oauth2/callback
-        - https://prometheus.apps.${cluster_name}.${base_domain}/oauth2/callback
+        - ${prometheus.domain}/oauth2/callback
```
Comment (mcanevet) [inline argocd/app-of-apps/values.tmpl.yaml:137]: same here

File: argocd/app-of-apps/values.tmpl.yaml [hunk 5]
```diff
@@ -130,13 +130,13 @@ keycloak:
...
         - https://alertmanager.apps.${base_domain}/oauth2/callback
-        - https://alertmanager.apps.${cluster_name}.${base_domain}/oauth2/callback
+        - ${alertmanager.domain}/oauth2/callback
```
Comment (mcanevet) [inline argocd/app-of-apps/values.tmpl.yaml:139]: same here

File: modules/argocd-helm/variables.tf [hunk 6]
```diff
@@ -51,13 +51,45 @@ variable "oidc" {
   }
 }
 
+variable "argocd" {
+  description = "ArgoCD settings"
+  type        = object({
+    domain = string
+  })
```
Comment (mcanevet) [inline modules/argocd-helm/variables.tf:58]: Shouldn't we add also `scheme` and `port` or just pass the `url` and extract `scheme`, `hostname` and `port` from it?

**Review by mcanevet — CHANGES_REQUESTED** [review]

File: argocd/app-of-apps/values.tmpl.yaml (hunk 1 above)
Comment (raphink) [reply argocd/app-of-apps/values.tmpl.yaml:65]: Interesting. It actually works this way in katacoda it seems

Comment (raphink) [reply argocd/app-of-apps/values.tmpl.yaml]: Fixed

File: modules/argocd-helm/variables.tf (hunk 6 above)
Comment (raphink) [reply modules/argocd-helm/variables.tf:58]: I think for now assuming https & 443 is reasonable.

**Comment by mcanevet** [comment]
Maybe we should use `host` or `hostname` instead of `domain` here ; the concept is not the same as `base_domain`.

**Pull request merged**